### ISSUE TITLE
How to make tauri use http://tauri.localhost/ instead of https://tauri.localhost/ ? The page was blocked since the page was loaded over HTTPS.

### ISSUE THREAD

**Issue by xuchaoqian** [tauri-apps/tauri#3007]
---
 
> **Updated @2023-09-17:** If anyone has encountered this problem, please pay attention to this PR: [Allow http origin on Windows](https://github.com/tauri-apps/tauri/pull/7645)

---

**Is your feature request related to a problem? Please describe.**
WebView console shows the following errors when we connect to endpoint 'ws:<URL>/': 
`Mixed Content: The page at '<URL>' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws:<URL>/'. This request has been blocked; this endpoint must be available over WSS.
`

**Describe the solution you'd like**
We should allow to set an option in tauri.conf.json to use http://tauri.localhost/ instead of https://tauri.localhost/.

**Describe alternatives you've considered**
None.

**Additional context**
1) We can NOT use WSS for historical reasons. 2) WS works fine on macOS, but does NOT works on Windows.

**Comment by lity** [comment]
I met the same problem too.

**Comment by mzy2240** [comment]
Me too. Electron supports this mode, and it is extremely useful for development and testing, even in some rare production env.

**Comment by amrbashir** [comment]
Can you post `tauri info` please ? and whether you added `ws:` to your csp configuration ?

Also please explain your feature request a bit further and how the api for it should look like, as I am having some trouble understanding your request.

**Comment by mzy2240** [comment]
I don't want to hijack the thread but here is mine:
```
Operating System - Windows, version 10.0.19044 X64
Webview2 - 96.0.1054.41

Node.js environment
  Node.js - 14.17.6
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - Not installed

Global packages
  npm - 6.14.15
  yarn - Not installed

Rust environment
  rustc - 1.56.1
  cargo - 1.56.0

App directory structure
/.git
/dist
/dist_electron
/node_modules
/public
/py
/pydist
/src
/src-tauri

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - Set automatically by Vue CLI plugin
  devPath - Set automatically by Vue CLI plugin
  framework - Vue.js (Vue CLI)
  bundler - Webpack
```
When running socketio client and connecting to a bundled python-based socketio server, I am facing exact same errors. Due to some reasons, only ws/http are accepted unfortunately. It seems to me that the only solution is somehow to support hosting over http. Please advise if you have any questions or comments.

Update: this is not a problem when just connecting to localhost, but becomes an issue when connecting from other ip.

**Comment by xuchaoqian** [comment]
Thank you for your participation.

> I don't want to hijack the thread but here is mine:
> 
> ```
> Operating System - Windows, version 10.0.19044 X64
> Webview2 - 96.0.1054.41
> 
> Node.js environment
>   Node.js - 14.17.6
>   @tauri-apps/cli - 1.0.0-beta.10
>   @tauri-apps/api - Not installed
> 
> Global packages
>   npm - 6.14.15
>   yarn - Not installed
> 
> Rust environment
>   rustc - 1.56.1
>   cargo - 1.56.0
> 
> App directory structure
> /.git
> /dist
> /dist_electron
> /node_modules
> /public
> /py
> /pydist
> /src
> /src-tauri
> 
> App
>   tauri.rs - 1.0.0-beta.8
>   build-type - bundle
>   CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
>   distDir - Set automatically by Vue CLI plugin
>   devPath - Set automatically by Vue CLI plugin
>   framework - Vue.js (Vue CLI)
>   bundler - Webpack
> ```
> 
> When running socketio client and connecting to a bundled python-based socketio server, I am facing exact same errors. Due to some reasons, only ws/http are accepted unfortunately. It seems to me that the only solution is somehow to support hosting over http. Please advise if you have any questions or comments.
> 
> Update: this is not a problem when just connecting to localhost, but becomes an issue when connecting from other ip.

**Comment by xuchaoqian** [comment]
> Can you post `tauri info` please ? and whether you added `ws:` to your csp configuration ?
> 
> Also please explain your feature request a bit further and how the api for it should look like, as I am having some trouble understanding your request.

### cargo tauri info:
```
Operating System - Windows, version 10.0.19042 X64
Webview2 - 96.0.1054.41

Node.js environment
  Node.js - 14.15.0
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 6.14.8
  yarn - 1.22.17

Rust environment
  rustc - 1.56.1
  cargo - 1.56.0

App directory structure
/.git
/bin
/build
/change-logs
/dist
/node_modules
/public
/src
/src-tauri
/src-web
/webpack

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../build
  devPath - http://localhost:9091
  framework - React
  bundler - Webpack
```

### The WebView console shows the following errors:
![image](https://user-images.githubusercontent.com/1064052/144530774-df5f96e4-df69-494a-befb-4fad9fce0fb0.png)

### Maybe the API looks like this:
    "security": {
      "useHttp": true,  // Add an option here, default value is false
      "csp": "default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'"
    }

**Comment by amrbashir** [comment]
Switching to `http` would probably disable secure context in the webview and some apis won't work.

**Comment by mzy2240** [comment]
Thank you for the heads up. Currently I'm only using the sidecar api, so hopefully it won't get affected, but I do hope we could have this option. Sometimes people turn to Tauri just because it could generate smaller and more performant app, instead of more native api supports.

**Comment by moxun33** [comment]
same problem to me.

**Comment by amrbashir** [comment]
> Thank you for the heads up. Currently I'm only using the sidecar api, so hopefully it won't get affected.

yeah it doesn't affect sidecar afaik.

Here is a [list](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts/features_restricted_to_secure_contexts) on MDN of features/apis that won't work if `window.isSecureContext` is `false`

I have a vague memory about `http` disabling secure context but I could be wrong so cc @lucasfernog

**Comment by mzy2240** [comment]
Does that mean this feature is already supported? If so, do you mind to explain how to enable it? If not, do you have plan to add the feature? Thanks

**Comment by lucasfernog** [comment]
I'll check this.

**Comment by lucasfernog** [comment]
Unfortunately we need the `https://tauri.localhost` URL to have a secure context enabled. Maybe this is a case where using a localhost is better. https://github.com/tauri-apps/tauri-plugin-localhost/

**Comment by mzy2240** [comment]
Thanks! Could we choose to disable the secure content? I'm trying to query a socketio server and due to some reasons it has to be either http or ws.

**Comment by spirityy** [comment]
I met the same problem.Because of historical reasons can not support wss,in this case windows compiler software becomes unusable, which should be a common situation.

> Unfortunately we need the `https://tauri.localhost` URL to have a secure context enabled. Maybe this is a case where using a localhost is better. https://github.com/tauri-apps/tauri-plugin-localhost/

hi, @lucasfernog 

This plugin needs my local yarn start the app and keep it running? I found that the source code can only run localhost, and I can't use other particular ip as a parameter?

**Comment by kmod-midori** [comment]
Based on the following sources:

Chrome:
https://chromestatus.com/feature/6269417340010496
https://bugs.chromium.org/p/chromium/issues/detail?id=589141
MDN:
https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
W3C:
https://w3c.github.io/webappsec-secure-contexts/#localhost

Both `https://tauri.localhost` and `http://tauri.localhost` should have `window.isSecureContext === true` if we are doing it correctly. Though this does not solve the mixed content error we are encountering there.

**Comment by lucasfernog** [comment]
> I met the same problem.Because of historical reasons can not support wss,in this case windows compiler software becomes unusable, which should be a common situation.
> 
> > Unfortunately we need the `https://tauri.localhost` URL to have a secure context enabled. Maybe this is a case where using a localhost is better. [tauri-apps/tauri-plugin-localhost](https://github.com/tauri-apps/tauri-plugin-localhost/)
> 
> hi, @lucasfernog
> 
> This plugin needs my local yarn start the app and keep it running? I found that the source code can only run localhost, and I can't use other particular ip as a parameter?

The plugin only applies to a production build. Why would you need a particular IP? Sorry for the lack of documentation, we're focusing on releasing tauri v1.0.0-rc.0 for now.

**Comment by lucasfernog** [comment]
> Based on the following sources:
> 
> Chrome: [chromestatus.com/feature/6269417340010496](https://chromestatus.com/feature/6269417340010496) [bugs.chromium.org/p/chromium/issues/detail?id=589141](https://bugs.chromium.org/p/chromium/issues/detail?id=589141) MDN: [developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) W3C: [w3c.github.io/webappsec-secure-contexts/#localhost](https://w3c.github.io/webappsec-secure-contexts/#localhost)
> 
> Both `https://tauri.localhost` and `http://tauri.localhost` should have `window.isSecureContext === true` if we are doing it correctly. Though this does not solve the mixed content error we are encountering there.

Yeah but that's not the case on Windows using Webview2.

**Comment by lucasfernog** [comment]
Maybe Webview2 is changing this soon? https://github.com/MicrosoftEdge/WebView2Feedback/issues/2102#issuecomment-1023759700

**Comment by wangyuxiangloveyou** [comment]
@lucasfernog tauri-plugin-localhost 可以支持程序cargo tauri build以后支持ws合http协议嘛   https://github.com/tauri-apps/tauri-plugin-localhost

**Comment by xiaowang-web** [comment]
@lucasfernog Can the Tauri plugin localhost program support WS and HTTP protocols after cargo Tauri build 
https://github.com/tauri-apps/tauri-plugin-localhost

**Comment by lucasfernog** [comment]
Yes it'll work @wangyuxiangloveyou

**Comment by lucasfernog** [comment]
Closing as the localhost plugin can be used to achieve this and we shouldn't allow insecure content by default.

**Comment by ChenArno** [comment]
> > Can you post `tauri info` please ? and whether you added `ws:` to your csp configuration ?
> > Also please explain your feature request a bit further and how the api for it should look like, as I am having some trouble understanding your request.
> 
> ### cargo tauri info:
> ```
> Operating System - Windows, version 10.0.19042 X64
> Webview2 - 96.0.1054.41
> 
> Node.js environment
>   Node.js - 14.15.0
>   @tauri-apps/cli - 1.0.0-beta.10
>   @tauri-apps/api - 1.0.0-beta.8
> 
> Global packages
>   npm - 6.14.8
>   yarn - 1.22.17
> 
> Rust environment
>   rustc - 1.56.1
>   cargo - 1.56.0
> 
> App directory structure
> /.git
> /bin
> /build
> /change-logs
> /dist
> /node_modules
> /public
> /src
> /src-tauri
> /src-web
> /webpack
> 
> App
>   tauri.rs - 1.0.0-beta.8
>   build-type - bundle
>   CSP - default-src blob: data: filesystem: ws: wss: http: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
>   distDir - ../build
>   devPath - http://localhost:9091
>   framework - React
>   bundler - Webpack
> ```
> 
> ### The WebView console shows the following errors:
> ![image](https://user-images.githubusercontent.com/1064052/144530774-df5f96e4-df69-494a-befb-4fad9fce0fb0.png)
> 
> ### Maybe the API looks like this:
> ```
> "security": {
>   "useHttp": true,  // Add an option here, default value is false
>   "csp": "default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'"
> }
> ```

Property useHttp is not allowed

**Comment by ChenArno** [comment]
> 是的，它可以工作@wangyuxiangloveyou

no,there http is not work

**Comment by spirityy** [comment]
> Maybe Webview2 is changing this soon? [MicrosoftEdge/WebView2Feedback#2102 (comment)](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2102#issuecomment-1023759700)

Any news update? we still need to use the plugin? cause the other platforms such as ubantu macos  were not need it.

**Comment by FabianLars** [comment]
@spirityy Well, you posted the answer to your question. In the [issues](https://togithub.com/MicrosoftEdge/WebView2Feedback/issues/172) linked in that comment it says that they will release it sometime soon.
But this change most likely won't make it into tauri until v2.

**Comment by Soya-xy** [comment]
Is there a solution?
When I use `tauri-plugin-localhost`  I am prompted that localhost cannot be accessed

**Comment by thinkloading** [comment]
请问你的问题解决了吗？自己开发的api，仅限ip访问，无法配置域名和ssl，现在还是v1版本

**Comment by spirityy** [comment]
> @spirityy Well, you posted the answer to your question. In the [issues](https://togithub.com/MicrosoftEdge/WebView2Feedback/issues/172) linked in that comment it says that they will release it sometime soon. But this change most likely won't make it into tauri until v2.

Okay. I think wv2 almost finished or close to be done.So tauri need to adapter it for v2? When will v2 be released?

**Comment by FabianLars** [comment]
> So tauri need to adapter it for v2?

Yes.

> When will v2 be released?

We don't know it yet. But the WebView2 feature implementation needs a little bit more time anyway: It's not implemented yet (knowing Microsoft's "speed" it probably will still take a while), and when it is released it will be in a prerelease version for a month and after that we need to wait another month to confidentiality use it in tauri.

**Comment by thinkloading** [comment]
好的，谢谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年7月13日(星期三) 晚上8:05
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tauri-apps/tauri] How to make tauri use http://tauri.localhost/ instead of https://tauri.localhost/ ? The page was blocked since the page was loaded over HTTPS. (Issue #3007)





  
所以tauri需要适配v2吗？
  
是的。
  
v2什么时候发布？
  
我们还不知道。但是 WebView2 的功能实现还是需要一点时间：它还没有实现（知道微软的“速度”可能还需要一段时间），当它发布时，它会在一个月后的预发布版本中我们需要再等一个月才能在tauri保密使用它。
 
—
直接回复此邮件，在 GitHub 上查看，或取消订阅。
您收到此消息是因为您发表了评论。消息 ID：<tauri-apps/tauri/issues/3007/1183137843 @ github 。com&gt;

**Comment by conkyliu** [comment]
my host is http,and websocker is ws,use devPath is dist file,  window run tairu dev or build  is failed, tip https://tauri.localhst/,how can i set config just use http not ssl.

**Comment by Richyeoh** [comment]
My host is http, and use websocket, but not supported wss, so what shall i do, supported ws protocol?

**Comment by xuchaoqian** [comment]
> Maybe Webview2 is changing this soon? [MicrosoftEdge/WebView2Feedback#2102 (comment)](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2102#issuecomment-1023759700)

@lucasfernog 
This feature is released, see here: [CoreWebView2CustomSchemeRegistration class](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2customschemeregistration?view=webview2-winrt-1.0.1369-prerelease)

Set property TreatAsSecure to true to allow us to use http protocol.

**Comment by FabianLars** [comment]
It's only available in pre-releases right now and at least in the win32 api it's marked as experimental.

> Set property TreatAsSecure to true to allow us to use http protocol.

Nope, it's not about that. It's about being a secure context; https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

**Comment by Soya-xy** [comment]
Thanks&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年7月13日(星期三) 晚上8:05
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tauri-apps/tauri] How to make tauri use http://tauri.localhost/ instead of https://tauri.localhost/ ? The page was blocked since the page was loaded over HTTPS. (Issue #3007)






所以 Tauri 需要把它适配到 v2 ？

是的。

v2 什么时候发布？

我们还不知道。但是 Web View 2 功能的实现还需要一点时间：它还没有实现（知道微软的“速度”它可能还需要一段时间），当它发布时，它将在一个月的预发布版本，之后，我们需要等待另一个一月的保密性使用它在金牛座。

—
直接回复此邮件，在 GitHub 上查看或取消订阅.
你收到这个是因为你评论了。留言 ID ***@***.***与&gt;

**Comment by ricardosc12** [comment]
Any update on the problem ? I still can't make requests using http

**Comment by learncodingforweb** [comment]
i am also facing same issue? Unable to use tauri-plugin-localhost. Any solution?

**Comment by LiChangyi** [comment]
I met the same problem too. I don't fetch `http protocol request`, It always responst status = 400 and the service is not receive request.

**Comment by UmashankarRahangdale** [comment]
@FabianLars Can you please guide me how can i make normal http calls using in a production build?.
Have gone through the previous discussion. Still not able to figure it out.

**Comment by huanlirui** [comment]
i am also facing same issue? Unable to use tauri-plugin-localhost. Any solution?

**Comment by huanlirui** [comment]
when i use  tauri-plugin-localhost , my application is  Flash back ,cant open

**Comment by megamew** [comment]
> when i use tauri-plugin-localhost , my application is Flash back ,cant open

me too

**Comment by jimmieW** [comment]
Hi, 

We are also facing this issue. I tried to create a fresh tauri app by following the quick start here:

https://tauri.app/v1/guides/getting-started/setup/

Then I tried integrating the plugin (https://github.com/tauri-apps/tauri-plugin-localhost) claims to go around the http limitation in **production**. 

When I am running it in **dev** (Windows 10) I am getting:

Finished dev [unoptimized + debuginfo] target(s) in 4.43s
Asset `index.html` not found; fallback to index.html.html
Asset `index.html` not found; fallback to index.html/index.html
Asset `index.html` not found; fallback to index.html
AssetNotFound("index.html")
Asset `index.html` not found; fallback to index.html.html
Asset `index.html` not found; fallback to index.html/index.html
Asset `index.html` not found; fallback to index.html
AssetNotFound("index.html")

<img width="402" alt="image" src="https://user-images.githubusercontent.com/10254358/214559104-1f2a9d91-b3eb-4976-a1ea-9cb6d09eb5dc.png">

I am using:
tauri = { version = "1.2", features = ["shell-open"] }

I tried "all" the versions available for the plugin (https://github.com/tauri-apps/tauri-plugin-localhost). Which one should be compatible with tauri 1.2?

Could we please get a minimal example where this plugin in being used in a "real" tauri app?

Thank you for your time!

**Comment by jimmieW** [comment]
For anyone else facing this issue. It does not work in debug but it actually works in release. 

A minimal example would probably still be useful for the community. 

Thanks

**Comment by learncodingforweb** [comment]
Still facing same problem, is there any update. I am using websocket to connect to localhost, after release, it tried to connect https. How can i connect to http in release mode

**Comment by Fredolx** [comment]
Any fix for this yet? Or we have to use localhost plugin?

**Comment by learncodingforweb** [comment]
Hi,
    Able to find solution for local server running on system and tauri app is running on same pc, where http server is installed. To make it work, need to change tauri.conf.json  
`"security": {  
      "csp": "default-src 'self' http://localhost:8000 ws://localhost:8000; img-src 'self' *; object-src 'none'; base-uri 'self'"  
    }  
`
Note, I am running http server on 8000 port.
**Tauri info as** 
Environment
  › OS: Debian 11 X64
  › Node.js: 16.13.2
  › npm: 8.1.2
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.2
  › rustc: 1.67.1
  › cargo: 1.67.1
  › Rust toolchain: stable-aarch64-unknown-linux-gnu 

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
  › @tauri-apps/api [NPM]: Not installed!
  › tauri [RUST]: 1.2.4,
  › tauri-build [RUST]: 1.2.1,
  › tao [RUST]: 0.15.8,
  › wry [RUST]: 0.23.4,

App
  › build-type: bundle
  › CSP: default-src 'self' http://localhost:8000 ws://localhost:8000; img-src 'self' *; object-src 'none'; base-uri 'self'
  › distDir: ../dist
  › devPath: http://localhost:4200/

App directory structure
  ├─ src-tauri
  ├─ ui
  ├─ dist
  └─ node_modules
Done in 16.63s.


If any one able to find solution for tauri app running on different pc, for me, it is not working even after changing **http://localhost:8000** with **http://ip_address:8000** please update so that other can also able to find solution

**Comment by mon** [comment]
For anyone struggling with the localhost plugin not working for `tauri dev`, here's a modified example that only enables it for release builds (works for me with vite):

`src-tauri/src/main.rs`
```rust
use tauri::{utils::config::AppUrl, window::WindowBuilder, WindowUrl};

// Tauri by default uses HTTPS, so use the localhost plugin to downgrade back to
// HTTP. However, when running tauri dev, we already have a localhost server
// provided by vite, so don't enable it
#[cfg(debug_assertions)]
const USE_LOCALHOST_SERVER: bool = false;
#[cfg(not(debug_assertions))]
const USE_LOCALHOST_SERVER: bool = true;

fn main() {
    let port = portpicker::pick_unused_port().expect("failed to find unused port");

    let window_url = if USE_LOCALHOST_SERVER {
        WindowUrl::External(format!("http://localhost:{}", port).parse().unwrap())
    } else {
        WindowUrl::App("index.html".into())
    };

    let mut context = tauri::generate_context!();
    let mut builder = tauri::Builder::default();

    if USE_LOCALHOST_SERVER {
        // rewrite the config so the IPC is enabled on this URL
        context.config_mut().build.dist_dir = AppUrl::Url(window_url.clone());
        context.config_mut().build.dev_path = AppUrl::Url(window_url.clone());
        builder = builder.plugin(tauri_plugin_localhost::Builder::new(port).build());
    }

    builder
        .setup(move |app| {
            WindowBuilder::new(app, "main".to_string(), window_url)
                .title("Localhost Example")
                .build()?;
            Ok(())
        })
        .run(context)
        .expect("error while running tauri application");
}
```

**Comment by prayag17** [comment]
I am still getting this error when I try to request a http address. How do I fix it?

**Comment by pot-code** [comment]
any updates? my app can send http request normally on windows, but immediately fail on macOS

**Comment by FabianLars** [comment]
@pot-code I am 99% sure your problem is unrelated to this issue. This one is windows-only and its fix won't touch the MacOS and Linux implementations at all.

My first idea would be that it may be related to the `exceptionDomain`, see the last point here: https://tauri.app/v1/guides/building/macos/#application-bundle-customization.
If this doesn't fix it please open a separate issue with a reproduction example.

**Comment by pot-code** [comment]
@FabianLars Thank you for your help! It is because I had set the `HTTP_PROXY` env variable that cause the request to fail, no clue why this would happen.

**Comment by icefee** [comment]
> when i use tauri-plugin-localhost , my application is Flash back ,cant open

[https://github.com/tauri-apps/plugins-workspace/issues/148](https://github.com/tauri-apps/plugins-workspace/issues/148#issuecomment-1692776095)

**Comment by trookie2000** [comment]
> > 当我使用 tauri-plugin-localhost 时，我的应用程序闪退，无法打开
> 
> [https://github.com/tauri-apps/plugins-workspace/issues/148](https://github.com/tauri-apps/plugins-workspace/issues/148#issuecomment-1692776095)

Maybe we could talk about it? I had a similar problem

**Comment by xuchaoqian** [comment]
If anyone has encountered this problem, please pay attention to this PR: [Allow http origin on Windows](https://github.com/tauri-apps/tauri/pull/7645)

**Comment by AceyKubbo** [comment]
> Any fix for this yet? Or we have to use localhost plugin?

i change the `tauri.conf.json > tauri > security > dangerousUseHttpScheme = true`,then it work

**Comment by conkyliu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by junior-slv** [comment]
> > Any fix for this yet? Or we have to use localhost plugin?
> 
> i change the `tauri.conf.json > tauri > security > dangerousUseHttpScheme = true`,then it work

you are an angel, thank you...

**Comment by conkyliu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by lvcj-1** [comment]
> > > 有解决这个问题的办法吗？还是我们必须使用 localhost 插件？
> > 
> > 
> > 我改变了`tauri.conf.json > tauri > security > dangerousUseHttpScheme = true`，然后它工作了
> 
> 你是天使，谢谢你……

为啥我的还是不行啊，我要碎了，不清楚我的配置是不是还需要修改什么
{
  "$schema": "../node_modules/@tauri-apps/cli/schema.json",
  "build": {
    "beforeBuildCommand": "npm run build:prod",
    "beforeDevCommand": "npm run dev",
    "devPath": "http://localhost:80",
    "distDir": "../dist",
    "withGlobalTauri": true
  },
  ...
  "tauri": {
  ...
  "security": {
      "csp": null,
      "dangerousUseHttpScheme" : true
    },
  ...
}

**Comment by AceyKubbo** [comment]
> > > > 有解决这个问题的办法吗？还是我们必须使用 localhost 插件？
> > > 
> > > 
> > > 我改变了`tauri.conf.json > tauri > security > dangerousUseHttpScheme = true`，然后它工作了
> > 
> > 
> > 你是天使，谢谢你……
> 
> 为啥我的还是不行啊，我要碎了，不清楚我的配置是不是还需要修改什么 { "$schema": "../node_modules/@tauri-apps/cli/schema.json", "build": { "beforeBuildCommand": "npm run build:prod", "beforeDevCommand": "npm run dev", "devPath": "http://localhost:80", "distDir": "../dist", "withGlobalTauri": true }, ... "tauri": { ... "security": { "csp": null, "dangerousUseHttpScheme" : true }, ... }

你可以看看allowlist有没有放行

**Comment by huaidanatu** [comment]
> dangerousUseHttpScheme = true

you're an Angel!  mua ,mua ,mua

**Comment by conkyliu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ljt019** [comment]
> > Any fix for this yet? Or we have to use localhost plugin?
> 
> i change the `tauri.conf.json > tauri > security > dangerousUseHttpScheme = true`,then it work

my goat

**Comment by conkyliu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by tdelabro** [comment]
tauri, is no longer an accepted field in tauri.conf.json. How do we allow this now?

**Comment by conkyliu** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by NikolaRHristov** [comment]
We have successfully resolved the Service Worker registration issues and file corruption problems:

1.  **Service Worker Fix**: Switched to using `tauri-plugin-localhost` with a dynamic port (Port 0) instead of trying to force `https://tauri.localhost`. This satisfies the browser's "Secure Context" requirement via `http://localhost`.
2.  **Runtime Config**: Updated `Mountain/Source/Binary.rs` to select a random port, inject the base URL into the window, and use `WebviewUrl::External`.
3.  **Frontend Logic**: Updated `Application/index.astro` to rely on `window.location.origin` (or injected globals) rather than hardcoded protocol checks.
4.  **File Corruption Fix**: Identified that conflicting `.editorconfig` settings (specifically `end_of_line`) were fighting with `rustfmt` inside VS Code, causing garbled text. We resolved this by unsetting EditorConfig rules for `*.rs` files.
